op-program/client/l2/engine.go

```diff
@@ -43,7 +41,13 @@ func (o *OracleEngine) L2OutputRoot(l2ClaimBlockNum uint64) (eth.Bytes32, error)
 	if err != nil {
 		return eth.Bytes32{}, fmt.Errorf("failed to open L2 state db at block %s: %w", outBlock.Hash(), err)
 	}
-	withdrawalsTrie, err := stateDB.Database().OpenStorageTrie(outBlock.Root, predeploys.L2ToL1MessagePasserAddr, stateDB.GetStorageRoot(predeploys.L2ToL1MessagePasserAddr))
+	storageRoot := stateDB.GetStorageRoot(predeploys.L2ToL1MessagePasserAddr)
+	storageTrie, err := stateDB.Database().OpenTrie(storageRoot)
+	if err != nil {
+		return eth.Bytes32{}, fmt.Errorf("opening storage trie: %w", err)
+	}
+	// TODO(Seb) confirm
+	withdrawalsTrie, err := stateDB.Database().OpenStorageTrie(outBlock.Root, predeploys.L2ToL1MessagePasserAddr, storageRoot, storageTrie)
```

Comment:
`OpenStorageTrie` now receives an additional `self Trie` as its last parameter that is returned directly in a Verkle tree case, AFAIU from the [internal cachingDB implementation](https://github.com/ethereum/go-ethereum/blob/566754c74a74c8175ec2f1ee5cc10a8caced6015/core/state/database.go#L191-L198).

Reviewers please confirm that this makes sense here. Can alternatively also pass `nil` in non-Verkle case.